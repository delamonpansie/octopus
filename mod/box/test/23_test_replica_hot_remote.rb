#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

class MasterEnv < RunEnv
  def test_root
    super << "_master"
  end

  def config
    super + <<EOD
wal_feeder_bind_addr = ":33034"
EOD
  end
end

class SlaveEnv < RunEnv
  def initialize
    @primary_port = 33023
    @test_root_suffix = "_slave"
    super
  end

  def config
    super + <<EOD
wal_feeder_addr = "127.0.0.1:33034"
local_hot_standby = 1
EOD
  end
end

master = MasterEnv.new.connect_eval do
  ping
  self
end

SlaveEnv.connect_eval do
  master.insert ['0', 'a', 'b', 'c', 'd']
  wait_for { select_nolog(['0']).length > 0 }


  master.select ['0']
  select ['0']
end

