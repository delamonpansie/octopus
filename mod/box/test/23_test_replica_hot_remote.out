# box.ping()
:pong

# box.insert(["0", "a", "b", "c", "d"])
1

# box.select(["0"])
[["0", "a", "b", "c", "d"]]

# box.select(["0"])
[["0", "a", "b", "c", "d"]]

